src/ddmd/root/file.d

```diff
@@ -77,12 +77,69 @@ nothrow:
     {
         if (len)
             return false; // already read the file
+
+        import core.stdc.string : strcmp;
+        const(char)* name = this.name.toChars();
+        if (strcmp(name, "-.d") == 0)
```

Comment:
Yes this is very ugly, but it's the simplest solution that produces the smallest diff to do what needs to be done.  The alternative is to make a whole bunch of complicated changes to `Module`'s ctor involving if-statements with complicated conditions to detect the `-` special case.  By just letting the `Module` ctor do what it currently does today, all we really need to do to make this work is to check for `"-.d"` instead of `-`.